docker/cuda.Dockerfile

```diff
@@ -35,25 +36,30 @@ ENV CUQUANTUM_DIR=/opt/nvidia/cuquantum
 # get required tools to build qsim
 RUN <<EOF
 python -m pip install pybind11
```

Comment:
It looks like this `pip install` happens outside of our venv. Should this line be moved down?

Reply:
Yes, AFAIU it's required only for that install so I chose to not pollute our venv with it (I might be mistaken though).

Reply:
@alexjuda after adapting to changes in https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/233, this is now being installed in the same virtualenv.